src/conversations/Conversations.ts

```diff
@@ -229,11 +229,11 @@ export default class Conversations {
       if (env.contentTopic === introTopic) {
         const messageBytes = b64Decode(env.message as unknown as string)
         const msg = await MessageV1.fromBytes(messageBytes)
-        await msg.decrypt(this.client.legacyKeys)
         const peerAddress = this.getPeerAddress(msg)
         if (!newPeer(peerAddress)) {
           return undefined
         }
+        await msg.decrypt(this.client.keystore, this.client.publicKeyBundle)
```

Comment:
Having to pass in the publicBundle here seems like its exposing complexity with minimal gain.

Today there is only a single PreKey, so the client gets the PublicBundle from the keyStore then it passes it back in.

In the future, there are multiple preKeys and the client will need to choose the correct matching bundle for the preKey which was used by the sender. Since this may not be the most current, the keyStore will need to provide access to multiple preKeys and then the client will need to choose which one to pass in.

In both these cases it seems that the KeyStore could perform all this work behind the scenes, and decrypt messages which it has keys for.

Reply:
The Keystore doesn't actually get the `PublicBundle` of the user. It just gets a boolean `isSender` flag, and the keys of the peer. This `decrypt` method just needs the client's bundle to know whether the user is the sender or the recipient to craft the right request to the Keystore.

Reply:
```proto
    // A single decryption request
    message Request {
        xmtp.message_contents.Ciphertext payload = 1;
        xmtp.message_contents.PublicKeyBundle peer_keys = 2;
        bytes header_bytes = 3;
        bool is_sender = 4;
    }
```

Reply:
Ah yes, traced it incorrectly 👍